packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -103,6 +99,12 @@ class ArtifactStore {
       targetPlatform: TargetPlatform.android
     ),
     const Artifact._(
+      name: 'Sky Shell',
```

Comment:
Yes, currently the binary on disk isn't marked executable, which is annoying.  Getting this in early will mean the next engine roll will mark it executable and we won't need to worry about handling that error case.